fred/cogs/crashes.py

```diff
@@ -300,16 +300,34 @@ async def process_file(self, file, extension) -> list[dict]():
                     return []
 
     async def mass_regex(self, text: str) -> AsyncIterator[dict]:
-        for crash in config.Crashes.fetch_all():
-            if match := await regex_with_timeout(crash["crash"], text, flags=re.IGNORECASE):
+
+        async def silly_async_gen():
+            nonlocal text
+            all_crashes = config.Crashes.fetch_all()
+            for crash_ in all_crashes:
+                yield crash_, regex_with_timeout(crash_["crash"], text, flags=regex.IGNORECASE)
+
+        async def patience(coro):
+            try:
+                return await coro
+            except asyncio.TimeoutError:
+                self.logger.warning("Regex timed out")
+
+        async for crash, queued_regex in silly_async_gen():
```

Comment:
I don't get why the async for is for. It won't execute concurrently, the behaviour is the same as previously

Reply:
It seemed faster when I tried it. That may have just been `regex` dunking on `re`. We can revert this if it doesn't actually benefit the runtime.

Reply:
It won't, it's just wrapping the old stuff in an async iterator and then iterating on the iterator so it's really the same behaviour (got got by that before :( it would be need if async fors could be run concurrently)